cmd/earth/main.go

```diff
@@ -320,13 +321,13 @@ func newEarthApp(ctx context.Context, console conslogging.ConsoleLogger) *earthA
 			Usage:       "The git password to use for git HTTPS authentication",
 			Destination: &app.gitPasswordOverride,
 		},
-		&cli.StringFlag{
-			Name:        "git-url-instead-of",
-			Value:       "",
-			EnvVars:     []string{"GIT_URL_INSTEAD_OF"},
-			Usage:       "Rewrite git URLs of a certain pattern. Similar to git-config url.<base>.insteadOf (https://git-scm.com/docs/git-config#Documentation/git-config.txt-urlltbasegtinsteadOf). Multiple values can be separated by commas. Format: <base>=<instead-of>[,...]. For example: 'https://github.com/=git@github.com:'",
-			Destination: &app.buildkitdSettings.GitURLInsteadOf,
-		},
+		//&cli.StringFlag{
+		//	Name:        "git-url-instead-of",
+		//	Value:       "",
+		//	EnvVars:     []string{"GIT_URL_INSTEAD_OF"},
+		//	Usage:       "Rewrite git URLs of a certain pattern. Similar to git-config url.<base>.insteadOf (https://git-scm.com/docs/git-config#Documentation/git-config.txt-urlltbasegtinsteadOf). Multiple values can be separated by commas. Format: <base>=<instead-of>[,...]. For example: 'https://github.com/=git@github.com:'",
+		//	Destination: &app.buildkitdSettings.GitURLInsteadOf,
+		//},
```

Comment:
Does our CI work if we remove this? I assume that with the HTTPS fallback it should work.

Wondering if we should still support something like this for `GIT CLONE` command. But perhaps if that's the only need, the user can always just use a build arg, so instead-of is no longer necessary.

Reply:
The fallback works for our CI; however, if someone is using this to inject some other auth, then that'll have to be updated to use the ~/.earthly/config.yml file. Maybe we can reach out to a few users to check in first to get an idea of how their CI is configured for auth.

Reply:
I'm thinking we should support commands like git has for editing config: `git config --global ...`. It'll help for configuring containerized CIs like GH actions, circle etc.

eg
```
earth config set ssh/github.com/ssh=https
```

or something like that. Anyway... that's a separate/future matter.

Reply:
I really like that idea! that'll be a great way to allow us to remove the depricated env flags once that's in place.